mlir/lib/Dialect/MIOpen/Tuning/GridwiseGemmParams.cpp

```diff
@@ -1283,6 +1139,22 @@ LogicalResult PopulateParamsXDL::isValidGemm(const InitParamsXDL &param,
   return success();
 }
 
+int64_t
+PopulateParamsXDL::calculatePaddingAmount(const InitParamsXDL &params,
+                                          const GemmSize &gemmSize) const {
+  int64_t padM = params.gemmMPerBlock - gemmSize.gemmM % params.gemmMPerBlock;
```

Comment:
That doesn't look right - I think you want padM to be 0 in the non-padding case. You probably want `mod_1_to_n` from the math utilities header here

Reply:
Yes, you are right -- reused the existing padding calculating utils there and refactored it a bit.